src/agent/main.cpp

```diff
@@ -230,7 +232,11 @@ void otPlatReset(otInstance *aInstance)
 
     gPlatResetReason = OT_PLAT_RESET_REASON_SOFTWARE;
 
-    otSysDeinit();
+    if (gApp)
```

Comment:
Is it possible that gApp is nullptr ?

Reply:
I think it's impossible among the use cases today. However, there's no harm to add a sanity check here.

Reply:
I think it's better to `Die` if such cases that are not supposed to happen happen.